src/routes/authentication.js

```diff
@@ -2,11 +2,22 @@ import express from 'express';
 
 import { agentLogin, Agentsignup, } from '../controllers/agent';
 
-import { validateUserSignup } from '../middleware';
+import {
+  agentProperties, allProperties, createProperty, deleteProperty, editProperty, getPropertyById
+} from '../controllers/propertyEndpoints/propertyendpoint';
+import { checkToken } from '../middleware/auth';
+
+import { validatePropertyInput, validateUserSignup } from '../middleware/middleware';
 
 const authenticationRouter = express.Router();
 
 authenticationRouter.post('/signup', validateUserSignup, Agentsignup);
 authenticationRouter.post('/login', agentLogin);
+authenticationRouter.get('/properties', allProperties);
+authenticationRouter.get('/property/:id', getPropertyById);
+authenticationRouter.post('/agent/properties', checkToken, validatePropertyInput, createProperty);
```

Comment:
Instead of `/agent/properties` this is better named `/property` you re creating a single resource and it's understandable that only an agent can create property.